### PR TITLE
Add warning regarding GTID consistency

### PR DESCRIPTION
**Pull request by bradjones1**
## This PR is a:

- [ ] New topic
- [ ] Content update
- [x] Content fix or rewrite
- [ ] Bug fix or improvement

## Summary

When this pull request is merged, it will add a warning to users of MySQL regarding Magento's dependency on a `TRIGGER` statement that is incompatible with GTID-based replication.

Refs https://github.com/magento/magento2/issues/15209

## Additional information

List all affected URLs 

- https://devdocs.magento.com/guides/v2.3/install-gde/prereq/mysql.html
- https://devdocs.magento.com/guides/v2.2/install-gde/prereq/mysql.html
- https://devdocs.magento.com/guides/v2.1/install-gde/prereq/mysql.html
- https://devdocs.magento.com/guides/v2.0/install-gde/prereq/mysql.html

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/devdocs?pullRequest=4129) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dev.magento.com/magento/devdocs?pullRequest=4129) before we can accept your contribution.<br/>

**Comment by magento-engcom-team** [comment]
@bradjones1 thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.0/install-gde/prereq/mysql.md [hunk 1]
```diff
@@ -33,6 +33,12 @@ Magento _strongly_ recommends you observe the following standard when you set up
 *	Familiarize yourself with [these potential MySQL trigger limitations](http://dev.mysql.com/doc/mysql-reslimits-excerpt/5.1/en/stored-program-restrictions.html){:target="_blank"} before you continue.
 *	If you use MySQL database replication, be aware that Magento does _not_ support MySQL statement-based replication. Make sure you use _only_ [row-based replication](http://dev.mysql.com/doc/refman/5.1/en/replication-formats.html){:target="_blank"}.
 
+{:.bs-callout .bs-callout-warning} Magento 2 currently utilizes `CREATE TEMPORARY TABLE` statements inside transactions,
```
Comment (erikmarr) [inline guides/v2.0/install-gde/prereq/mysql.md]: We are no longer making changes to 2.0 content. Please remove this file from your PR.

File: guides/v2.1/install-gde/prereq/mysql.md [hunk 2]
```diff
@@ -30,6 +30,12 @@ Magento _strongly_ recommends you observe the following standard when you set up
 *	Familiarize yourself with [these potential MySQL trigger limitations](http://dev.mysql.com/doc/mysql-reslimits-excerpt/5.1/en/stored-program-restrictions.html){:target="_blank"} before you continue.
 *	If you use MySQL database replication, be aware that Magento does _not_ support MySQL statement-based replication. Make sure you use _only_ [row-based replication](http://dev.mysql.com/doc/refman/5.1/en/replication-formats.html){:target="_blank"}.
 
+{:.bs-callout .bs-callout-warning} Magento 2 currently utilizes `CREATE TEMPORARY TABLE` statements inside transactions,
+which are [incompatible](https://dev.mysql.com/doc/refman/5.7/en/replication-gtids-restrictions.html) with database
+implementations utilizing GTID-based replication.
+(E.g., [Google Cloud SQL second-generation instances](https://cloud.google.com/sql/docs/features#differences).) See
```
Comment (erikmarr) [inline guides/v2.1/install-gde/prereq/mysql.md]: Please remove the link to the github issue. We do not link to specific issues in devdocs (except for the release notes).

File: guides/v2.2/install-gde/prereq/mysql.md [hunk 3]
```diff
@@ -26,6 +26,12 @@ Magento _strongly_ recommends you observe the following standard when you set up
 *	Familiarize yourself with [these potential MySQL trigger limitations](http://dev.mysql.com/doc/mysql-reslimits-excerpt/5.1/en/stored-program-restrictions.html){:target="_blank"} before you continue.
 *	If you use MySQL database replication, be aware that Magento does _not_ support MySQL statement-based replication. Make sure you use _only_ [row-based replication](http://dev.mysql.com/doc/refman/5.1/en/replication-formats.html){:target="_blank"}.
 
+{:.bs-callout .bs-callout-warning} Magento 2 currently utilizes `CREATE TEMPORARY TABLE` statements inside transactions,
+which are [incompatible](https://dev.mysql.com/doc/refman/5.7/en/replication-gtids-restrictions.html) with database
+implementations utilizing GTID-based replication.
+(E.g., [Google Cloud SQL second-generation instances](https://cloud.google.com/sql/docs/features#differences).) See
```
Comment (erikmarr) [inline guides/v2.2/install-gde/prereq/mysql.md]: See above about linking to specific issues.

File: guides/v2.3/install-gde/prereq/mysql.md [hunk 4]
```diff
@@ -26,6 +26,12 @@ Magento _strongly_ recommends you observe the following standard when you set up
 *	Familiarize yourself with [these potential MySQL trigger limitations](http://dev.mysql.com/doc/mysql-reslimits-excerpt/5.1/en/stored-program-restrictions.html){:target="_blank"} before you continue.
 *	If you use MySQL database replication, be aware that Magento does _not_ support MySQL statement-based replication. Make sure you use _only_ [row-based replication](http://dev.mysql.com/doc/refman/5.1/en/replication-formats.html){:target="_blank"}.
 
+{:.bs-callout .bs-callout-warning} Magento 2 currently utilizes `CREATE TEMPORARY TABLE` statements inside transactions,
+which are [incompatible](https://dev.mysql.com/doc/refman/5.7/en/replication-gtids-restrictions.html) with database
+implementations utilizing GTID-based replication.
+(E.g., [Google Cloud SQL second-generation instances](https://cloud.google.com/sql/docs/features#differences).) See
```
Comment (erikmarr) [inline guides/v2.3/install-gde/prereq/mysql.md]: See above about linking to specific issues.

**Comment by bradjones1** [comment]
@erikmarr Thanks for the review. Changes made per request.

File: guides/v2.1/install-gde/prereq/mysql.md [hunk 5]
```diff
@@ -30,6 +30,11 @@ Magento _strongly_ recommends you observe the following standard when you set up
 *	Familiarize yourself with [these potential MySQL trigger limitations](http://dev.mysql.com/doc/mysql-reslimits-excerpt/5.1/en/stored-program-restrictions.html){:target="_blank"} before you continue.
 *	If you use MySQL database replication, be aware that Magento does _not_ support MySQL statement-based replication. Make sure you use _only_ [row-based replication](http://dev.mysql.com/doc/refman/5.1/en/replication-formats.html){:target="_blank"}.
 
+{:.bs-callout .bs-callout-warning} Magento 2 currently utilizes `CREATE TEMPORARY TABLE` statements inside transactions,
+which are [incompatible](https://dev.mysql.com/doc/refman/5.7/en/replication-gtids-restrictions.html) with database
+implementations utilizing GTID-based replication.
+(E.g., [Google Cloud SQL second-generation instances](https://cloud.google.com/sql/docs/features#differences).)
```
Comment (keharper) [inline guides/v2.1/install-gde/prereq/mysql.md]: Global for all versions:

Change "replication. (E.g.," to "replication, such as "

Comment (bradjones1) [reply guides/v2.1/install-gde/prereq/mysql.md]: Done.

**Comment by bradjones1** [comment]
@erikmarr Thanks. Though it looks like the warning formatting isn't applying? https://devdocs.magento.com/guides/v2.2/install-gde/prereq/mysql.html

**Comment by erikmarr** [comment]
Thanks for letting me know! I'm looking into it now.

**Pull request merged**